usaspending_api/search/v2/views/spending_by_category_views/base_spending_by_category.py

```diff
@@ -124,39 +123,42 @@ def common_db_query(self, queryset: QuerySet, django_filters: dict, django_value
             .order_by("-amount")
         )
 
-    def elasticsearch_results_generator(self, filter_query: ES_Q, size: int = 1000) -> list:
-        """
-        The number of buckets required across the different categories varies. To solve this a generator is created
-        with the assumption that the "apply_elasticsearch_aggregations" implementation makes use of partitions.
-        This will allow for any number of results to be retrieved regardless of the number of buckets.
-        """
-        bucket_count = get_number_of_unique_terms(filter_query, self.category.primary_field)
-        num_partitions = (bucket_count // size) + 1
-
-        for partition in range(num_partitions):
-            search = self.build_elasticsearch_search_with_aggregations(filter_query, partition, num_partitions, size)
-            response = search.handle_execute()
-            if response is None:
-                raise Exception("Breaking generator, unable to reach cluster")
-            results = self.build_elasticsearch_result(response.aggs.to_dict())
-            yield results
-
-    def query_elasticsearch(self, filter_query: ES_Q) -> list:
-        results = self.elasticsearch_results_generator(filter_query)
-        flat_results = list(itertools.chain.from_iterable(results))
-        return flat_results
-
-    @abstractmethod
-    def build_elasticsearch_search_with_aggregations(
-        self, filter_query: ES_Q, curr_partition: int, num_partitions: int, size: int
-    ) -> TransactionSearch:
+    def build_elasticsearch_search_with_aggregations(self, filter_query: ES_Q) -> Optional[TransactionSearch]:
         """
         Using the provided ES_Q object creates a TransactionSearch object with the necessary applied aggregations.
         All aggregations are applied in a specific order to insure the correct nested aggregations.
 
         The top-most aggregation is expected to use partitions to make sure that all buckets are retrieved.
         """
-        pass
+        # Create the filtered Search Object
+        search = TransactionSearch().filter(filter_query)
+
+        # Get count of unique buckets; terminate early if there are no buckets matching criteria
+        bucket_count = get_number_of_unique_terms(filter_query, f"{self.category.agg_field}.hash")
+        if bucket_count == 0:
```

Comment:
Logging a warning and throwing an error that notifies a user to use more filters should this case come up. As of now this case should not occur, more of a just in case.